test_util/test_runner.py

```diff
@@ -3,18 +3,30 @@
 Note: ssh_user must be able to use docker without sudo priveleges
 """
 import logging
+import tempfile
 import time
 from contextlib import contextmanager
 from multiprocessing import Process
 from os.path import join
-from subprocess import CalledProcessError, TimeoutExpired
+from subprocess import CalledProcessError, TimeoutExpired, check_call
 
 import pkg_resources
 
+import gen.template
+
 LOGGING_FORMAT = '[%(asctime)s|%(name)s|%(levelname)s]: %(message)s'
 logging.basicConfig(format=LOGGING_FORMAT, level=logging.DEBUG)
 log = logging.getLogger(__name__)
 
+BASH_TEMPLATE = """
+#!/bin/bash
```

Comment:
I'd add a `set -euo pipefail; set -x` here to make it harder for us to make bash mistakes and miss them.